lib/internal/Magento/Framework/GraphQl/Query/Resolver/ResolveLogger.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\GraphQl\Query\Resolver;
+
+use GraphQL\Error\ClientAware;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritDoc
+ */
+class ResolveLogger implements ResolveLoggerInterface
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $clientLogger;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $serverLogger;
+
+    /**
+     * @param LoggerInterface $clientLogger
+     * @param LoggerInterface $serverLogger
+     */
+    public function __construct(
```

Comment:
Please take a look at a possibility to use Factory for creating the needed logger - passing two params with the same type looks suspicious according to the Igor Miniailo.
Thx!